### DISCUSSION TITLE
Is it possible to use models like DALL-E 2/3 or TTS-1 with Copilot for Obsidian?

### DISCUSSION THREAD

**Discussion by novanoticia** [logancyang/obsidian-copilot#923] [Q&A]
Hi,
First of all, thank you for this fantastic plugin! I’ve successfully integrated all text-based models from ChatGPT, which is greatly appreciated. However, I have some questions regarding compatibility with image generation and other content types:
	1.	Is it possible to use models like DALL-E 2, DALL-E 3, or TTS-1 with Copilot for Obsidian?
	2.	My current setup is as follows:
	•	Copilot version: 2.7.9
	•	Obsidian version: 1.7.7
	•	I have a ChatGPT Pro account with available API credits.
	3.	I’ve tried integrating these models, but I couldn’t get them to work (although all text-based models work perfectly, which is great!).
	4.	Is it possible to use other providers or configurations to generate images, such as third-party APIs?
	5.	Are there any tutorials or videos available that explain how to configure these models in the plugin?

I’d appreciate any information or guidance on this topic. Thanks in advance!

---------------
Hola
Primero, gracias por este fantástico plugin, ya que he logrado integrar sin problemas todos los modelos de texto de ChatGPT, lo cual es de agradecer. Sin embargo, tengo algunas dudas respecto a la compatibilidad con modelos de generación de imágenes y otros tipos de contenido
	1.	¿Es posible utilizar modelos como DALL-E 2, DALL-E 3 o TTS-1 con Copilot for Obsidian?
	2.	Mi configuración actual es la siguiente:
	•	Versión de Copilot: 2.7.9
	•	Versión de Obsidian: 1.7.7
	•	Tengo una cuenta Pro de ChatGPT con créditos disponibles para su API
	3.	He intentado integrar estos modelos, pero no he conseguido que funcionen (mientras que los modelos de texto funcionan perfectamente)
	4.	¿Es posible utilizar otros proveedores o configuraciones para generar imágenes, como por ejemplo utilizando APIs de terceros?
	5.	¿Existe algún tutorial o video que explique cómo configurar este tipo de modelos en el plugin?

Agradecería cualquier información o guía sobre este tema. ¡Muchas gracias de antemano!

**Comment by logancyang** [comment]
Hi @novanoticia thanks for using the plugin!

Image generation and TTS are on my roadmap, once they are supported I will post a video demo on [my youtube channel](https://www.youtube.com/@loganhallucinates?sub_confirmation=1)! Please consider subscribing to not miss the update!

**Reply by novanoticia** [reply]
That's fantastic, its plugins are very well documented, it has barely cost me work to add the API of all the models of all providers, there is less left.

Thank you very much again!
-------
Eso es fantástico, está muy bien documentada su plugins, apenas me ha costado trabajo el añadir la API de todos los modelos de todos los proveedores, ya falta menos. 
Muchas gracias de Nuevo!

**Reply by danrfletcher** [reply]
@logancyang I know it's been a while since TTS was mentioned as a roadmap item. I'd like to build it as a contribution rather than wait on it. I posted a concrete proposal on [issue #995](https://github.com/logancyang/obsidian-copilot/discussions/link) for a native TTS provider with playback, highlighting, and speed control on chat messages. Would appreciate your thoughts on whether that's something you'd merge before I put time into it.
